actix-http/src/body/boxed.rs

```diff
@@ -8,10 +8,15 @@ use std::{
 use bytes::Bytes;
 
 use super::{BodySize, MessageBody, MessageBodyMapErr};
-use crate::Error;
 
 /// A boxed message body with boxed errors.
-pub struct BoxBody(Pin<Box<dyn MessageBody<Error = Box<dyn StdError>>>>);
+pub struct BoxBody(BoxBodyInner);
+
+enum BoxBodyInner {
+    None,
```

Comment:
i kinda think this should wrap `body::None` and delegate so that the behavior is guaranteed to be the same